### PR TITLE
Highlight me on timeline -- better

### PR DESCRIPTION
**Pull request by EvilOne**
A better way to highlight _me_ on the timeline.

Comments are welcome. I want to get the idea out there as I look to clean things up a bit more.

### CODE REVIEW

File: Packages/Timeline/Sources/Timeline/TimelineViewModel.swift [hunk 1]
```diff
@@ -152,6 +154,15 @@ extension TimelineViewModel: StatusesFetcher {
                                                                   maxId: nil,
                                                                   minId: nil,
                                                                   offset: statuses.count))
+      if let account {
```
Comment (Dimillian) [inline Packages/Timeline/Sources/Timeline/TimelineViewModel.swift:157]: I'm ok with the logic, should be refactored into a proper function like processStatuses or something, later we'll probably want to add more local stuff.

File: Packages/Models/Sources/Models/Status.swift [hunk 2]
```diff
@@ -59,6 +59,8 @@ public struct Status: AnyStatus, Decodable, Identifiable, Equatable, Hashable {
     id + createdAt + (editedAt ?? "")
   }
 
+  public var shouldHighlight: Bool?
```
Comment (Dimillian) [inline Packages/Models/Sources/Models/Status.swift:62]: To separate that a bit, I would make a new protocol like protocol `StatusUI` and add this the props inside. Then make `Status` conform to this new protocol. Name the property something like `ui_shouldHighlight`. I know it's not pretty but that way it should be crystal clear what is backend / models status and UI extensions for the app.

File: Packages/Timeline/Sources/Timeline/TimelineView.swift [hunk 3]
```diff
@@ -75,6 +75,7 @@ public struct TimelineView: View {
       if viewModel.client == nil {
         viewModel.client = client
         viewModel.timeline = timeline
+        viewModel.account = account
```
Comment (Dimillian) [inline Packages/Timeline/Sources/Timeline/TimelineView.swift:78]: You don't need to pass the Current account env here. You can directly use `CurrentAccount.shared`, it's fine as a singleton.

File: Packages/Timeline/Sources/Timeline/TimelineViewModel.swift (hunk 1 above)
Comment (EvilOne) [reply Packages/Timeline/Sources/Timeline/TimelineViewModel.swift:157]: Totally agree! On my list before remove the draft status.

File: Packages/Models/Sources/Models/Status.swift (hunk 2 above)
Comment (EvilOne) [reply Packages/Models/Sources/Models/Status.swift:62]: I like this better than what I have. Will try to come up with a good variable name.

File: Packages/Timeline/Sources/Timeline/TimelineView.swift (hunk 3 above)
Comment (EvilOne) [reply Packages/Timeline/Sources/Timeline/TimelineView.swift:78]: Missed the singleton. Thanks!

**Pull request closed without merging**